### DISCUSSION TITLE
Coral edge TPUs being available again and under $50 could we take advantage of this?

### DISCUSSION THREAD

**Discussion by rhamblen** [oobabooga/text-generation-webui#4379] [Ideas]
With tpu's being much lower in price, it is easy to build more powerful hardware using edge TPUs.  There is a board which takes multiple coral processors increasing the capabilities to silly levels of power.

Is it worth building an option to support this co-processor too?

https://coral.ai/
https://coral.ai/products/
https://www.asus.com/networking-iot-servers/aiot-industrial-solutions/gpu-edge-ai-accelerators/ai-accelerator-pcie-card/

**Comment by rhamblen** [comment]
OK so I realise the models for Tensor Flow are different.  BUT I found two articals which can help here...
1.  Converting pytorch models to Tensorflow
https://discuss.tensorflow.org/t/conversion-of-pytorch-code-to-tensorflow/15468
2. Converting pytorch models to Tensoflow lite (which is needed for coral edge TPUs.
https://www.kaggle.com/code/jarvisai7/tensorflow-pytorch-to-tflite

The process seems to be 
pytorch to intermediary (using Onix or  Nobuco) to Tensorflow THEN Tensorflow to Tensorflow Lite

But I've not done any MI coding (yet) so I dont know the implications of all this, other than it is in theory possible.

To my point.... if a TPU is super quick at very low cost (capital and power), if you can run multiple in a PC, then it may have an advantage over a high end GPU.
